src/schema/text_options.rs

```diff
@@ -9,6 +9,7 @@ use std::ops::BitOr;
 pub struct TextOptions {
     indexing: Option<TextFieldIndexing>,
     stored: bool,
+    tokenized: bool,
```

Comment:
Here goes the annoying comment, that will require extra work... :)

The problems with putting this on `TextOptions` is that it makes it possible to represent nonsensical TextOptions.
For instance, what is the point of a text field that is not indexed but is pretokenized?

For this reasons, the field tokenizer is defined in `TextFieldIndexing`. 
Here are alternative options that are better, and we can discuss
- Declare the concept of pretokenized as a tokenizer (I did not investigate the implication in the rest of the code)
- Remove all change in the schema. If the document carries tokens, then use those and ignore the configure tokenizer  <----- I like this solution better.

Reply:
If you are concerned mainly with the nonsensical TextOptions I think we could simply make pretokenized field being always indexed as well i.e. `PRETOKENIZED` `TextOptions` would simply be `TEXT` with `pretokenized` set to `true`.
Making API simpler is a good argument, but there are some things which we should consider. 1) Taking away indication from the schema that the field is pre-tokenized will make us guess the `Value` type( pretokenized vs text) from JSON when parsing and 2) this allows for fields with mixed` PretokenizedString` and `String` values.
I think both of these are not really a problem (but maybe I miss something), so we can go with eliminating the flag from `TextOptions`.

Reply:
> If you are concerned mainly with the nonsensical TextOptions I think we could simply make pretokenized field being always indexed as well i.e. PRETOKENIZED TextOptions would simply be TEXT with pretokenized set to true.

I won't discuss the first solution, since we both seem to agree the second one (removing pretokenized from the schema) is better. Let's go for this then.

> Taking away indication from the schema that the field is pre-tokenized will make us guess the Value type( pretokenized vs text) 

Yes. An alternative solution would be to have a Text Value type that optionally have a `pretokenized: Option<TokenizedText>`. Either way the deserialization/serialization logic needs to be a bit smarter.